QuickFIXn/Store/FileStore.cs

```diff
@@ -198,13 +194,19 @@ public void Get(SeqNumType startSeqNum, SeqNumType endSeqNum, List<string> messa
     {
         for (SeqNumType i = startSeqNum; i <= endSeqNum; i++)
         {
-            if (_offsets.ContainsKey(i))
+            if (_offsets.TryGetValue(i, out MsgDef msgDef))
             {
-                _msgFile.Seek(_offsets[i].Index, System.IO.SeekOrigin.Begin);
-                byte[] msgBytes = new byte[_offsets[i].Size];
-                _msgFile.Read(msgBytes, 0, msgBytes.Length);
-
-                messages.Add(CharEncoding.SelectedEncoding.GetString(msgBytes));
+                _msgFile.Seek(msgDef.Index, System.IO.SeekOrigin.Begin);
+                byte[] msgBytes = ArrayPool<byte>.Shared.Rent(msgDef.Size);
+                try
+                {
+                    _ = _msgFile.Read(new Span<byte>(msgBytes, 0, msgDef.Size));
```

Comment:
I know it is unrelated to this PR, but this Read call should not ignore the return value

```suggestion
                    _msgFile.ReadExactly(new Span<byte>(msgBytes, 0, msgDef.Size));
```